### PR TITLE
Add a route sections postprocessor 

### PR DESCRIPTION
**Pull request by vyskocil**
The route sections postprocessor give information on the via point and section length between when computing route with via point between start and target.
The demo Routing application has been updated to take a optional --via parameter : 

```
% ./Routing --via 48.8571 2.3295 ~/Documents/OSM/France.osmscout 48.85690 2.34136 48.85934 2.33663
No speed for type 'highway_residential_area' defined!
No speed for type 'highway_service_area' defined!
No speed for type 'highway_track' defined!
Using 'CalculateRouteViaCoords' method
From:                48.85700 N 2.34143 E Way 565610149[1 *> 3759113579648971777]
To:                  48.85736 N 2.32890 E Way 565570763[3759113583642588161 >* 2]
Time:                0.000
Air-line distance:   0.9 km
Minimum cost:        29 s
Actual cost:         00:04:17
Cost limit:          00:14:38
Route nodes loaded:  176
Route nodes ignored: 350
Max. OpenList size:  25
Max. ClosedSet size: 176
Warning: Index intersections.idx has cache size 10000, but requires cache size 57825 to load index completely into cache!
From:                48.85736 N 2.32890 E Way 565570763[2 *> 3759113583642588161]
To:                  48.85968 N 2.33526 E Way 565604655[3759113613854439681 >* 4]
Time:                0.000
Air-line distance:   0.6 km
Minimum cost:        19 s
Actual cost:         00:02:37
Cost limit:          00:13:22
Route nodes loaded:  122
Route nodes ignored: 227
Max. OpenList size:  20
Max. ClosedSet size: 122
Warning: Index intersections.idx has cache size 10000, but requires cache size 57825 to load index completely into cache!
Loading path data: 0.000
Scanning locations: 0.032
Loading candidates: 0.032
Analysing candidates: 0.000
Postprocessing time: 0.033
----------------------------------------------------
     At| After|  Time| After|
----------------------------------------------------
  0.0km         0:00h        Start at 'Start'
                             Drive along highway_residential 'Place du Pont Neuf'
  0.0km  0.0km  0:00h  0:00h At crossing 'Place du Pont Neuf', 'Pont Neuf', 'Quai de l'Horloge'
                             Turn left into highway_residential 'Quai de l'Horloge'
  0.0km  0.0km  0:00h  0:00h Way changes name from 'Quai de l'Horloge' to 'Place du Pont Neuf'
  0.2km  0.0km  0:00h  0:00h Way changes name from 'Place du Pont Neuf' to 'Pont Neuf'
  0.2km  0.0km  0:00h  0:00h At crossing 'Pont Neuf', 'Quai de Conti', 'Quai des Grands Augustins', 'Rue Dauphine', 'Rue de Nevers'
                             Turn right into highway_primary 'Quai de Conti'
  0.5km  0.0km  0:01h  0:00h Way changes name from 'Quai de Conti' to 'Quai Malaquais'
  0.8km  0.0km  0:02h  0:00h At crossing 'Quai Malaquais', 'Quai Voltaire', 'Rue des Saints-Pères'
                             Turn left into highway_tertiary 'Rue des Saints-Pères'
  1.0km  0.0km  0:02h  0:00h At crossing 'Rue Jacob', 'Rue de l'Université', 'Rue des Saints-Pères'
                             Turn right into highway_residential 'Rue de l'Université'
  1.3km  0.1km  0:03h  0:00h Etap: 1
  1.3km  0.1km  0:03h  0:00h At crossing 'Rue de l'Université', 'Rue du Bac'
                             Turn right into highway_secondary 'Rue du Bac'
  1.4km  0.1km  0:03h  0:00h Pass: Jean Louis David
  1.6km  0.0km  0:03h  0:00h At crossing 'Pont Royal', 'Quai Valéry Giscard d'Estaing', 'Quai Voltaire', 'Rue du Bac'
                             Turn slightly right into highway_secondary 'Pont Royal'
  1.7km  0.1km  0:04h  0:00h At crossing 'Pont Royal'
                             Turn right into highway_secondary 'Pont Royal'
  1.7km  0.0km  0:04h  0:00h Way changes name from 'Pont Royal' to 'Quai François Mitterrand'
  2.1km         0:04h        Target reached 'Target'
Description generation time: 0.000
```

### CODE REVIEW

File: Demos/src/Routing.cpp [hunk 1]
```diff
@@ -741,6 +751,13 @@ int main(int argc, char* argv[])
                       "penalty-max",
                       "Maximum junction penalty, time [s]. Default "s + std::to_string(duration_cast<seconds>(args.maxPenalty).count()));
 
+
+  argParser.AddOption(osmscout::CmdLineGeoCoordOption([&args](const osmscout::GeoCoord& value) {
+                        args.via=value;
```
Comment (Karry) [inline Demos/src/Routing.cpp:756]: It would be great to accept --via argument multiple times, so make args.via vector of locations

Comment (vyskocil) [reply Demos/src/Routing.cpp:756]: Yes I agree but I don’t know how to code it

Comment (Framstag) [reply Demos/src/Routing.cpp:756]: (Without checking the details...). The Command Line Argument Parser does not hold state to error on arguments called multiple times. In this case it should be just as easy as making args.via a list and do a ...push_back() in the callback implementation!?

Comment (vyskocil) [reply Demos/src/Routing.cpp:756]: Yes it works, thank you ! I made a pull request with that

**Pull request merged**